core/templates/pages/contributor-dashboard-page/contributor-badge/contributor-badge.component.ts

```diff
@@ -0,0 +1,294 @@
+        this.totalTranslationStats[languageDescription].corrections += (
+          stat.accepted_translations_with_reviewer_edits_count);
+      });
+    }
+    if (allContributionStats.question_contribution_stats.length > 0) {
+      allContributionStats.question_contribution_stats.map((stat) => {
+        this.totalQuestionStats.submissions += stat.accepted_questions_count;
+      });
+    }
+    if (allContributionStats.question_review_stats.length > 0) {
+      allContributionStats.question_review_stats.map((stat) => {
+        this.totalQuestionStats.reviews += stat.reviewed_questions_count;
+        this.totalQuestionStats.corrections += (
+          stat.accepted_questions_with_reviewer_edits_count);
+      });
+    }
+
+    for (let language in this.totalTranslationStats) {
+      this.translationBadges[language] = {};
+
+      this.translationBadges[language][
+        AppConstants.CONTRIBUTION_STATS_SUBTYPE_SUBMISSION] =
+          this.getObtainedBadges(
+            this.totalTranslationStats[language].submissions,
+            AppConstants.CONTRIBUTION_STATS_SUBTYPE_SUBMISSION,
+            this.totalTranslationStats[language].language);
+      this.translationBadges[language][
+        AppConstants.CONTRIBUTION_STATS_SUBTYPE_REVIEW] =
+          this.getObtainedBadges(
+            this.totalTranslationStats[language].reviews,
+            AppConstants.CONTRIBUTION_STATS_SUBTYPE_REVIEW,
+            this.totalTranslationStats[language].language);
+      this.translationBadges[language][
+        AppConstants.CONTRIBUTION_STATS_SUBTYPE_CORRECTION] =
+          this.getObtainedBadges(
+            this.totalTranslationStats[language].corrections,
+            AppConstants.CONTRIBUTION_STATS_SUBTYPE_CORRECTION,
+            this.totalTranslationStats[language].language);
+    }
+
+    if (this.userCanSuggestQuestions) {
+      this.questionSubmissionBadges = this.getObtainedBadges(
+        this.totalQuestionStats.submissions,
+        AppConstants.CONTRIBUTION_STATS_SUBTYPE_SUBMISSION);
+    }
+    if (this.userCanReviewQuestionSuggestions) {
+      this.questionReviewBadges = this.getObtainedBadges(
+        this.totalQuestionStats.reviews,
+        AppConstants.CONTRIBUTION_STATS_SUBTYPE_REVIEW);
+      this.questionCorrectionBadges = this.getObtainedBadges(
+        this.totalQuestionStats.corrections,
+        AppConstants.CONTRIBUTION_STATS_SUBTYPE_CORRECTION);
+    }
+
+    this.languages = Object.keys(this.totalTranslationStats);
+
+    if (this.languages.length > 0) {
+      this.selectedLanguage = this.languages[0];
+      this.userCanReviewTranslationSuggestion = (
+        this.reviewableLanguages.includes(this.selectedLanguage));
+    }
+    this.dataLoading = false;
+    console.log(this.translationBadges);
+    console.log(this.questionSubmissionBadges);
+  }
+
+  getObtainedBadges(
+      contributionCount: number,
+      contributionSubType: string, language?: string): Badge[] {
```

Comment:
```suggestion
      contributionCount: number,
      contributionSubType: string, 
      language?: string
  ): Badge[] {
```

Why is language optional?

Reply:
We create Question contribution badges from this function as well. They dont have an associated language. Hence it is an optional param here.

Reply:
Done